axelrod/strategies/backstabber.py

```diff
@@ -17,52 +17,81 @@ class BackStabber(Player):
     classifier = {
         'memory_depth': float('inf'),
         'stochastic': False,
-        'makes_use_of': set(['length']),
+        'makes_use_of': {'length'},
         'long_run_time': False,
         'inspects_source': False,
         'manipulates_source': False,
         'manipulates_state': False
     }
 
-    @staticmethod
-    def strategy(opponent: Player) -> Action:
-        if not opponent.history:
-            return C
-        if opponent.defections > 3:
-            return D
-        return C
+    def strategy(self, opponent: Player) -> Action:
+        return _backstabber_strategy(opponent)
 
 
-@FinalTransformer((D, D), name_prefix=None) # End with two defections
+@FinalTransformer((D, D), name_prefix=None)  # End with two defections
 class DoubleCrosser(Player):
     """
     Forgives the first 3 defections but on the fourth
-    will defect forever. If the opponent did not defect
-    in the first 6 rounds the player will cooperate until
-    the 180th round. Defects on the last 2 rounds unconditionally.
+    will defect forever. Defects on the last 2 rounds unconditionally.
+
+    If 8 <= current round <= 180,
+    if the opponent did not defect in the first 7 rounds,
+    the player will only defect after the opponent has defected twice in-a-row.
     """
 
     name = 'DoubleCrosser'
     classifier = {
         'memory_depth': float('inf'),
         'stochastic': False,
-        'makes_use_of': set(['length']),
+        'makes_use_of': {'length'},
         'long_run_time': False,
         'inspects_source': False,
         'manipulates_source': False,
         'manipulates_state': False
     }
 
     def strategy(self, opponent: Player) -> Action:
-        cutoff = 6
-
-        if not opponent.history:
-            return C
-        if len(opponent.history) < 180:
-            if len(opponent.history) > cutoff:
-                if D not in opponent.history[:cutoff + 1]:
-                    if opponent.history[-2:] != [D, D]:  # Fail safe
-                        return C
-        if opponent.defections > 3:
-            return D
+        if _opponent_triggers_alt_strategy(opponent):
+            return _alt_strategy(opponent)
+        return _backstabber_strategy(opponent)
+
+
+def _backstabber_strategy(opponent: Player) -> Action:
+    """
+    Cooperates until opponent defects a total of four times, then always defects.
+    """
+    if not opponent.history:
         return C
+    if opponent.defections > 3:
+        return D
+    return C
+
+
+def _alt_strategy(opponent: Player) -> Action:
```

Comment:
Could you write specific tests for these module functions please.

The don't need to be extensive but just enough so that they'd fail if something got refactored in the strategy methods removing the need for them.